mbed_tools_ci/assert_news.py

```diff
@@ -42,133 +40,64 @@ def validate_file_contents(self) -> None:
-                ) or self.git.is_release_branch(self.current_branch)
-
-    def verify(self) -> None:
-        """Checks that news files were added in the current branch as part of the PR's changes.
-
-        The files are then individually checked in order to ensure
-        they follow the standard in terms of naming and content.
-
-        """
-        if not self.is_feature_branch():
-            logger.info(
-                f'No need for news file on branch [{self.current_branch}]'
-            )
-            return
-        added_news = self.find_news_file()
-        news_dir = configuration.get_value(ConfigurationVariable.NEWS_DIR)
-        project_root = configuration.get_value(
-            ConfigurationVariable.PROJECT_ROOT)
-        if not added_news or len(added_news) == 0:
-            raise FileNotFoundError(
-                f'PR must contain a news file in {news_dir}. See README.md'
-            )
-        logger.info(
-            f'{len(added_news)} new news files found in `{news_dir}`'
+def validate_news_files(git: GitWrapper, news_dir: str) -> None:
+    """Checks that news files exist and pass validation checks."""
+    added_news_files = find_news_files(git=git, news_dir=news_dir)
+    if not added_news_files:
+        raise FileNotFoundError(
+            f"PR must contain news file in {news_dir}. See README.md."
         )
-        logger.info(':: Checking news files format')
-        for news_file in added_news:
-            NewsFileValidator(os.path.realpath(
-                os.path.join(project_root, news_file))).validate()
+    for news_file in added_news_files:
+        validate_news_file(news_file)
 
 
 def main() -> None:
     """Asserts the new PR comprises at least one news file and it adheres to the required standard.
 
     An exception is raised if a problem with the news file is found.
     """
-    parser = argparse.ArgumentParser(
-        description='Publish target data report to AWS.')
-    parser.add_argument("-v", "--verbose", action="count", default=0,
-                        help="Verbosity, by default errors are reported.")
+    parser = argparse.ArgumentParser(description="Publish target data report to AWS.")
+    parser.add_argument(
+        "-v",
+        "--verbose",
+        action="count",
+        default=0,
+        help="Verbosity, by default errors are reported.",
+    )
     args = parser.parse_args()
     set_log_level(args.verbose)
+    news_dir = configuration.get_value(ConfigurationVariable.NEWS_DIR)
+    project_root = configuration.get_value(ConfigurationVariable.PROJECT_ROOT)
+    relative_news_dir = pathlib.Path(news_dir).relative_to(project_root)
     try:
-        NewsFileDiscoverer().verify()
+        validate_news_files(git=GitWrapper(), news_dir=str(relative_news_dir))
     except Exception as e:
         log_exception(logger, e)
         sys.exit(1)
 
 
-if __name__ == '__main__':
+if __name__ == "__main__":
```

Comment:
Has this file been "blacked"?

Reply:
Don't know what you're talking about. 😉 Is it a problem? I'd probably make quotes consistent manually if it wasn't `black`.